app/src/main/java/de/westnordost/streetcomplete/quests/QuestUtil.kt

```diff
@@ -81,3 +81,12 @@ fun getNameOrBrandOrOperatorOrRef(tags: Map<String, String>): String? {
         else -> null
     }
 }
+
+fun getRef(tags: Map<String, String>): String? {
+    val ref = tags["ref"]
+
+    return when {
+        ref != null -> ref
+        else -> null
+    }
```

Comment:
If you are concerned about accessing the right tag all the time, you could make a `const val REF_TAG = "ref"`.

Reply:
> If you are concerned about accessing the right tag all the time, you could make a const val REF_TAG = "ref".

That sounds like `const val TWO = 2`

Reply:
Indeed, it is a little silly. But only a little: with a constant, the compiler will warn you about spelling mistakes :)

Reply:
> This could be replaced by just return ref. In fact, the whole getRef function call should be replaced by tags["ref"].

Right, I missed that it would be so simple. Limiting to ref was a good idea, making getRef function was not.

Reply:
> This could be replaced by just `return ref`. In fact, the whole getRef function call should be replaced by `tags["ref"]`.

uhm well, sounds too simple :D

Reply:
It's not the right thing to do here, but *in general* another option is using kotlin's [delegated properties](https://kotlinlang.org/docs/delegated-properties.html#storing-properties-in-a-map).

```kt
val ref: String by tags
// equivalent to
val ref: String get() = tags["ref"]
```